### PR TITLE
feat(testing): Add testing infrastructure with Kover and test utilities

### LINKED ISSUE #3516: Proposal: Improve Test Coverage Infrastructure
Assalamu Alaikum,

I'm a new contributor interested in helping improve the test coverage for this project. After exploring the codebase, I noticed there's an opportunity to strengthen the testing foundation, which could help prevent regressions and make future contributions safer.

### Current State

- ~24 test files primarily in the `app` module
- No code coverage tooling configured
- Limited shared test utilities

### Proposal

I'd like to contribute a testing infrastructure improvement in phases:

**Phase 1: Foundation (Ready)**
- Add Kover plugin for code coverage reporting
- Create `common:test-utils` module with shared test infrastructure:
  - `TestDispatcherRule` for coroutines testing
  - `RxSchedulerRule` for RxJava testing
  - Flow test extensions (Turbine-based)
  - `TestDataFactory` for Quran-specific test fixtures
- Expand `QuranInfo` tests from 6 to 46 tests (boundary conditions, invalid inputs, edge cases)

**Future Phases (if Phase 1 is welcomed)**
- Phase 2: Model layer tests (SuraAyah, bookmarks)
- Phase 3: Repository/data layer tests
- Phase 4: Presenter/ViewModel tests
- Phase 5: Integration tests

### Why This Approach

1. **Non-breaking**: No changes to production code
2. **Incremental**: Small, reviewable PRs
3. **Foundation-first**: Utilities that make future testing easier
4. **Follows existing patterns**: Matches project's code style and architecture

### Questions for Maintainers

Before I submit a PR, I wanted to check:

1. Is improving test coverage something you'd welcome?
2. Any preferences on testing frameworks? (I used JUnit4, Truth, MockK, Turbine based on existing tests)
3. Should the `test-utils` module live under `common/` or elsewhere?
4. Any specific areas you'd prioritize for testing?

I have Phase 1 ready locally and can submit a PR whenever you're ready to review. Happy to adjust the approach based on your feedback.

JazakAllahu Khairan for maintaining this beautiful project.

### PR DESCRIPTION
**Pull request by ksalhab89**
## Summary

Implements Phase 1 of the testing infrastructure initiative as discussed in #3516.

- Add Kover code coverage plugin with proper exclusions for generated code
- Create `common:test-utils` module with shared testing utilities
- Add `TestDataFactory` using real domain models (SuraAyah, Bookmark, Tag, RecentPage)
- Add `RxSchedulerRule` for synchronous RxJava testing
- Add `TestDispatcherRule` for coroutine testing
- Add Flow testing extensions with Turbine
- Remove MockK dependency (prefer fakes over mocks per maintainer feedback)
- Fix pre-existing broken tests in audio modules
- Add comprehensive `TESTING_STRATEGY.md` documentation

## Changes

| Commit | Description |
|--------|-------------|
| feat(testing) | Add Kover plugin and test-utils module |
| test(core) | Add comprehensive tests for QuranInfo |
| refactor(testing) | Remove MockK, prefer fakes |
| refactor(testing) | Improve test infrastructure quality |
| fix(tests) | Fix pre-existing broken tests in audio modules |

## Test Results

- **128 tests passing**, 0 failures
- All modules compile and lint checks pass
- Kover coverage report generates successfully

## Test Plan

- [x] All existing tests pass
- [x] New QuranInfo tests pass
- [x] Fixed audio module tests pass
- [x] Lint checks pass
- [x] Kover report generates

Closes #3516

### CODE REVIEW

**Review by ahmedre — APPROVED** [review]
جزاكم الله خيراً looks good
2 minor comments

**Comment by ahmedre** [comment]
would also be nice to kill Mockito in the future in sha' Allah

File: common/test-utils/src/main/kotlin/com/quran/labs/test/TestDataFactory.kt [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    sura = null,
+    ayah = null,
+    page = page,
+    timestamp = timestamp
+  )
+
+  /**
+   * Creates multiple bookmarks for testing.
+   */
+  fun createBookmarks(count: Int): List<Bookmark> =
```
[the remark this answers is not in the document]
Comment (ksalhab89) [reply common/test-utils/src/main/kotlin/com/quran/labs/test/TestDataFactory.kt]: Good catch\! You're right - removed for now. Will add back in Phase 1 when implementing BookmarksDaoImpl tests (15+ tests). Fixed in 4e1dc3c.

**Comment by ksalhab89** [comment]
## Future Work: Mockito Removal

Planning to remove Mockito as part of Phase 2 and/or 3, in sha' Allah.

**Current state:**
- 7 test files using Mockito (~100 mock calls)
- Mocking: Context, Resources, QuranSettings, BookmarksDBAdapter, models

**Strategy:**
- Phase 2: Create fakes naturally while writing new presenter/model tests (30+ tests)
- Phase 2/3: Migrate existing tests to use fakes instead of mocks
- Aligns with "fakes over mocks" philosophy

This approach ensures fakes are built with real use cases rather than upfront guesswork.
